canopeum_frontend/src/pages/SiteSocialPage.tsx

```diff
@@ -139,8 +139,18 @@ const SiteSocialPage = () => {
         <div className='row row-gap-1 m-0'>
           <div className='col-12 col-md-6 col-lg-5 col-xl-4'>
             <div className='d-flex flex-column gap-4'>
-              <AnnouncementCard announcement={site.announcement} viewMode={viewMode} />
-              <ContactCard contact={site.contact} viewMode={viewMode} />
+              <AnnouncementCard
+                announcement={site.announcement}
+                viewMode={viewMode}
+                onEdit={(editedAnnoucement: Announcement) =>
+                  setSite(site => ({ ...site, announcement: editedAnnoucement } as SiteSocial))}
+              />
+              <ContactCard
+                contact={site.contact}
+                viewMode={viewMode}
+                onEdit={(editedContact: Contact) =>
+                  setSite(site => ({ ...site, contact: editedContact } as SiteSocial))}
```

Comment:
Unless it shadows a variable, you can shorten to
```suggestion
                onEdit={(announcement: Announcement) =>
                  setSite(site => ({ ...site, announcement} as SiteSocial))}
              />
              <ContactCard
                contact={site.contact}
                viewMode={viewMode}
                onEdit={(contact: Contact) =>
                  setSite(site => ({ ...site, contact} as SiteSocial))}
```

Type should be infered from the assigned parameter, it's likely you can do:

```suggestion
                onEdit={announcement => setSite(site => ({ ...site, announcement} as SiteSocial))}
              />
              <ContactCard
                contact={site.contact}
                viewMode={viewMode}
                onEdit={contact => setSite(site => ({ ...site, contact} as SiteSocial))}
```

I'm not sure about that cast either.